packages/localization/src/dates/modifyDateBy.ts

```diff
@@ -7,32 +7,49 @@ import CalendarDate from "./CalendarDate.js";
  * @param date CalendarDate instance
  * @param amount how many days/months/years to add (can be a negative number)
  * @param unit what to modify: "day", "month" or "year"
+ * @param isPageUpOrPageDown whether the method is called when the date is being modified by the PageUp/PageDown keys/logic
  * @param minDate minimum date to enforce
  * @param maxDate maximum date to enforce
  */
-const modifyDateBy = (date: CalendarDate, amount: number, unit: string, minDate?: CalendarDate, maxDate?: CalendarDate) => {
+const modifyDateBy = (date: CalendarDate, amount: number, unit: string, isPageUpOrPageDown?:boolean, minDate?: CalendarDate, maxDate?: CalendarDate) => {
 	const newDate = new CalendarDate(date);
-	if (unit === "day") {
+
+	switch (unit) {
+	case "day":
 		newDate.setDate(date.getDate() + amount);
-	} else if (unit === "month") {
-		newDate.setMonth(date.getMonth() + amount);
-		const stillSameMonth = amount === -1 && newDate.getMonth() === date.getMonth(); // f.e. PageUp remained in the same month
-		const monthSkipped = amount === 1 && newDate.getMonth() - date.getMonth() > 1; // f.e. PageDown skipped a whole month
-		if (stillSameMonth || monthSkipped) { // Select the last day of the month in any of these 2 scenarios
-			newDate.setDate(0);
+		break;
+	case "month":
+		if (isPageUpOrPageDown) { // In cases where the method is called when the date is being modified by the PageUp/PageDown keys/logic
+			newDate.setMonth(date.getMonth() + amount);
+			const stillSameMonth = amount === -1 && newDate.getMonth() === date.getMonth(); // f.e. PageUp remained in the same month
+			const monthSkipped = amount === 1 && newDate.getMonth() - date.getMonth() > 1; // f.e. PageDown skipped a whole month
+			if (stillSameMonth || monthSkipped) { // Select the last day of the month in any of these 2 scenarios
```

Comment:
I think that was from the old logic, but I have questions if this is expected.

In two cases we return set last day of the month as the comment suggests "Select the last day of the month in any of these 2 scenarios"
- stillSameMonth // f.e. PageUp remained in the same month
- monthSkipped // f.e. PageDown skipped a whole month
if (stillSameMonth || monthSkipped) {
newDate.setDate(0);
}

Is "newDate.setDate(0);" setting the last day?

Should we really set the last day of the month if we pressed PageUP and remained in the same month (stillSameMonth). 

Somehow it looks counterintuitive. With PageUp, the user tries to get one month back (e.g. amount is -1) and if in some case when "stillSameMonth" is true and the date is part of the current month - we set it to the last day if we believe the comment "Select the last day of the month in any of these 2 scenarios". 

I would imagine we set it at least to the first because we pressed PageUp.

Could you check this?

Reply:
Answered offline - no change required regarding this comment

Reply:
As discussed offline - this logic is for the cases where min/maxDate is/are set.